tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTestBase.java

```diff
@@ -131,26 +117,62 @@ String checkStatusReadAndCloseResponse(Response.Status expectedStatus, Response
     /**
      * The started LRA will be named based on the class name and the running test name.
      */
-    protected String lraClientId() {
+    String lraClientId() {
         return this.getClass().getSimpleName() + "#" + testName.getMethodName();
     }
 
     /**
      * Adjusting the default timeout by the specified timeout factor
      * which can be defined by user.
      */
-    protected long lraTimeout() {
+    long lraTimeout() {
         return Util.adjust(LraTckConfigBean.LRA_TIMEOUT_MILLIS, config.timeoutFactor());
     }
 
-
-    void replayEndPhase(String resource) {
-        // trigger a recovery scan which trigger a replay attempt on any participants
+    /**
+     * Trigger or wait for a recovery scan to replay the protocol termination phase (complete or compenstate)
+     * on a participants that is in need of recovery.
+     *
+     * @param resource the resource on which recovery should be replayed. If null then recovery
+     *                should be attempted on all eligible participants
+     */
+    void replayEndPhase(String resource) throws InterruptedException {
+        // trigger a replay attempt on any participants
         // that have responded to complete/compensate requests with Response.Status.ACCEPTED
-        WebTarget recoveryPath = recoveryTarget.path("recovery");
-        Response response2 = recoveryPath
-                .request().get();
+        int recoveryPasses = 2; // 1 for the normal end call and 2 for the recovery pass call
```

Comment:
default value not useful. `recoveryPasses` gets his value from `checkStatusReadAndCloseResponse()`(line 159) and loop is controlled by time or when `checkStatusReadAndCloseResponse()` returns 0.

So probably also use another name.

Reply:
The variable `recoveryPasses` is used to tell the resource how many times it should return `202 ACCEPTED` from the completion call on line 144 and the initial value is used. This second usage of the variable that you refer to is the resource saying how many more times it is expecting to be asked to complete and when that value hits zero we know that recovery has happened.

In other words the variable has the same semantic meaning in both usages (it is keeping track of the number completion calls: the client sets the initial value and the resource decrements it) so using the same name makes sense.

Reply:
@rdebusscher So can I resolve this conversation?

Reply:
I added more documentation explaining how the method operates